spec/models/miq_report_result_spec.rb

```diff
@@ -1,14 +1,56 @@
 describe MiqReportResult do
-  context "persisting generated report results" do
-    before(:each) do
-      EvmSpecHelper.local_miq_server
+  before :each do
+    EvmSpecHelper.local_miq_server
+
+    @user1 = FactoryGirl.create(:user_with_group)
+  end
+
+  context "report result created by User 2 with current group 2" do
+    before :each do
+      @report = FactoryGirl.create(:miq_report)
+      group_2 = FactoryGirl.create(:miq_group)
+      @user1.miq_groups << group_2
+      @report_result1 = FactoryGirl.create(:miq_report_result, :miq_report_id => @report.id, :miq_group => group_2)
+      @report_result2 = FactoryGirl.create(:miq_report_result, :miq_report_id => 999_999_999)
```

Comment:
Why hardcode 999_999_999?  I would think you can test all combinations like so:

a) 1 report with 2 results
b) 1 report with 1 result
c) 1 result with nil miq_report_id (doesn't get returned ever)

1) Make sure you get a's 2 results with a's report.id, not c's result
2) Make sure you don't b when passing no report.id